packages/experiments/README.md

```diff
@@ -7,33 +7,42 @@ This package acts as a "dealer" that only allows WordPress packages to use the e
 Each package needs to start by registering itself:
 
 ```js
-const { register } =
-	__dangerousOptInToUnstableAPIsOnlyForCoreModules(
-		'<CONSENT STRING>',  // See index.js, this may change without notice.
-		'@wordpress/blocks'
-	);
+const { lock, unlock } = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+	'<CONSENT STRING>', // See index.js, this may change without notice.
+	'@wordpress/blocks'
+);
 ```
 
 The function name communicates that plugins are not supposed to use it. To make double and triple sure, the first argument must be that exact consent string, and the second argument must be a known `@wordpress` package that hasn't opted in yet – otherwise an error is thrown.
 
-Expose a new `__experimental` API as follows:
+The `lock` and `unlock` API may be used as follows:
 
 ```js
-export const __experiments = register( { __unstableGetInnerBlocksProps } )
-```
-
-Consume the registered `__experimental` APIs as follows:
-
-```js
-// In the @wordpress/block-editor package:
-import { __experiments as blocksExperiments } from '@wordpress/blocks';
-const { unlock } =
-	__dangerousOptInToUnstableAPIsOnlyForCoreModules(
-		'<CONSENT STRING>',  // See index.js
-		'@wordpress/block-editor'
-	);
-
-const { __unstableGetInnerBlocksProps } = unlock( blocksExperiments );
+// In the package exposing the experimental APIs:
+const { lock } = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+	'<CONSENT STRING>', // See index.js, this may change without notice.
+	'@wordpress/blocks'
+);
+export const __experiments = {};
+lock( __experiments, {
+	privateFunction() {
+		console.log( 'I am a private function!' );
+	},
+} );
+
+// In the package consuming the experimental APIs:
+const { unlock } = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+	'<CONSENT STRING>', // See index.js, this may change without notice.
+	'@wordpress/block-editor'
```

Comment:
I'm probably not understanding how this work but I thought that to "unlock" '@wordpress/blocks' APIs, I need to provide the same package name here?

Reply:
@youknowriad The code you highlighted is a deterrent to only deal the `lock()` and `unlock()` functions to core WordPress packages. By passing the `@wordpress/block-editor` package name, I'm saying "This is `@wordpress/block-editor`" opting in, do not accept this name again". In other words, I couldn't do this:

```js
// Only the first opt-in for a given package name works:
_dangerousOptInToUnstableAPIsOnlyForCoreModules(
	'<CONSENT STRING>',
	'@wordpress/block-editor'
);

// Any subsequent attempt throws an exception:
_dangerousOptInToUnstableAPIsOnlyForCoreModules(
	'<CONSENT STRING>',
	'@wordpress/block-editor'
);
```

Reply:
The idea is to hide private data in public objects via `lock()`, and retrieve it via `unlock()` as follows:

```js
// In @wordpress/package1:
export const publicObject = {}; 
lock(publicObject, 'Private string');

// In @wordpress/package2:
import { publicObject } from '@wordpress/package1';

console.log(unlock(publicObject));
// "Private string"

console.log(publicObject);
// Just "{}", you can't directly retrieve the private string from that object
```